framework/src/problems/FEProblemBase.C

```diff
@@ -5508,18 +5508,30 @@ FEProblemBase::computeResidualTags(const std::set<TagID> & tags)
 
   try
   {
-    _aux->compute(EXEC_LINEAR);
+    try
```

Comment:
@lindsayad Any suggestion? The code here appears to properly catch the exception, but only in serial. In parallel, the simulation just hangs. I thought the `PARALLEL_CATCH` macros you put together around `ComputeMortarNodalAuxBndThread<AuxKernel>` would take care of that.

Reply:
PARALLEL_CATCH only performs a mpi communication of `_has_exception` and `_exception _message` and the rethrows on every processor. For that to work there has t be a try/catch before the parallel catch that calls setExceptionMessage (or whatever it's called)